src/v/kafka/server/materialized_partition.h

```diff
@@ -12,11 +12,15 @@
 #include "kafka/server/partition_proxy.h"
 #include "storage/log.h"
 
+#include <seastar/core/shared_ptr.hh>
+
 namespace kafka {
 class materialized_partition final : public kafka::partition_proxy::impl {
 public:
-    explicit materialized_partition(storage::log log)
-      : _log(log) {}
+    explicit materialized_partition(
```

Comment:
maybe we should have a separate metric for materialized partition ? I have this solved with virtualized interface for `partition_probe` this way we can provide different implementation for materialized partition

Reply:
Have you already written a solution for this? If so I don't want to step on your toes here